biz.aQute.bndlib/src/aQute/bnd/build/ProjectLauncher.java

```diff
@@ -143,13 +110,57 @@ public ProjectLauncher(Project project) throws Exception {
 	 * @throws Exception
 	 */
 	protected void updateFromProject() throws Exception {
-		setCwd(project.getBase());
+		setCwd(getProject().getBase());
 
 		// pkr: could not use this because this is killing the runtests.
-		// project.refresh();
+		// getProject().refresh();
 		runbundles.clear();
 		startlevels.clear();
-		Collection<Container> run = project.getRunbundles();
+
+		getProject().getBundles(Strategy.HIGHEST, Constants.DEFAULT_LAUNCHER_BSN, null)
```

Comment:
This must go to the ProjectLauncherImpl since it is only valid there. This Project Launcher is abstract. I also think it needs its own method and use other methods. I actually worked on the same thing :-(

Reply:
For the record, and as we discussed, this was fixed by https://github.com/bndtools/bnd/pull/3318!